render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,344 @@
+                }).catch((error: Error) => {
+                    Service.setPortAvailable(this.port.path, false);
+                    this._ng_isAvailable = false;
+                    console.log(error.message);
+                });
+                this._forceUpdate();
+            }
+        } else {
+            console.error('Something went wrong while loading the session');
+        }
+    }
+
+    private _formatLoad(load: number): string {
+        let read: string = '';
+        if (load > 1024 * 1024 * 1024) {
+            read = (load / 1024 / 1024 / 1024).toFixed(2) + ' Gb';
+        } else if (load > 1024 * 1024) {
+            read = (load / 1024 / 1024).toFixed(2) + ' Mb';
+        } else if (load > 1024) {
+            read = (load / 1024).toFixed(2) + ' Kb';
+        } else {
+            read = load + ' b';
+        }
+        return read;
+    }
+
+    private _createChart() {
+        this._ctx = this.canvas.nativeElement.getContext('2d');
+        const cSessionPort = this._getSessionPort();
+        if (cSessionPort) {
+            const cSparklineData = cSessionPort.sparkline_data;
+            this._chart = new Chart(this._ctx, {
+                type: 'line',
+                data: {
+                    labels: this._chart_labels.slice(0, Service.chart_limit),
+                    datasets: [{
+                        data: cSparklineData.slice(cSparklineData.length - Service.chart_limit),
+                        borderColor: Service.getPortColor(this.port.path),
+                        pointRadius: 0,
+                        fill: false,
+                    }]
+                },
+                options: {
+                    maintainAspectRatio: false,
+                    animation: {
+                        duration: 0
+                    },
+                    scales: {
+                        xAxes: [{
+                            display: false,
+                        }],
+                        yAxes: [{
+                            display: false,
+                            stacked: true,
+                            ticks: {
+                                beginAtZero: true,
+                            },
+                            gridLines: {
+                                drawOnChartArea: false
+                            }
+                        }]
+                    },
+                    legend: {
+                        display: false
+                    },
+                    tooltips: {
+                        enabled: false
+                    }
+                }
+            });
+        } else {
+            console.error('Something went wrong with the SessionPort entry');
+        }
+    }
+
+    private _getSessionPort(): IPort {
+        const cSessionPort = Service.sessionPort[this._session];
+        if (cSessionPort && cSessionPort[this.port.path]) {
+            return cSessionPort[this.port.path];
+        }
+        return null;
+    }
```

Comment:
you can return `null`, but return declaration has only `IPort`
Probably should be corrected to:
```
    private _getSessionPort(): IPort | null { ... }
```